src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapFandom.java

```diff
@@ -1,206 +1,230 @@
+    AutoplayVuap videoFanTakeover = prepareSlot(slotName, fandomPage);
+
+    videoFanTakeover.play();
+    videoFanTakeover.close();
+    videoFanTakeover.waitForVideoPlayerHidden();
+  }
+
+  @Test(
+          dataProviderClass = FandomAdsDataProvider.class,
+          dataProvider = "vuapPage",
+          groups = {"AdsVuapFandomDesktop", "AdsVuapCheckSlotSizesFandom"}
+  )
+  public void adsVuapCheckSlotSizesFandom(String pageType, String pageName, String slotName) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    AutoplayVuap videoFanTakeover = prepareSlot(slotName, fandomPage);
+
+    videoFanTakeover.waitForAdToLoad();
+    double imageHeight = videoFanTakeover.getAdSlotHeight();
+
+    videoFanTakeover.play();
+    videoFanTakeover.waitForVideoStart();
+
+    double videoHeight = videoFanTakeover.getAdVideoHeight();
+    Assertion.assertTrue(VuapAssertions.isVideoAdBiggerThanImageAdOasis(videoHeight, imageHeight));
+
+    videoFanTakeover.waitForVideoPlayerHidden();
+    Assertion.assertTrue(VuapAssertions.isImageAdInCorrectSize(videoFanTakeover));
+  }
+
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+          dataProviderClass = FandomAdsDataProvider.class,
+          dataProvider = "vuapPage",
+          groups = {"AdsVuapFandomDesktop", "AdsVuapTimeProgressDesktopFandom"}
+  )
+  public void adsVuapTimeProgressingFandom(String pageType, String pageName, String slotName) throws InterruptedException {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    AutoplayVuap videoFanTakeover = prepareSlot(slotName, fandomPage);
+
+    networkTrafficInterceptor.startIntercepting();
+
+    videoFanTakeover.playVideoFor(Duration.ofSeconds(1));
+    double firstPause = videoFanTakeover.getCurrentTime();
+
+    videoFanTakeover.playVideoFor(Duration.ofSeconds(1));
+    double currentTime = videoFanTakeover.getCurrentTime();
+
+    Assert.assertTrue(
+            firstPause < currentTime,
+            String.format(
+                    "Video time is not progressing, first pause time %s is not smaller than current %s",
+                    firstPause, currentTime)
+    );
+  }
+
+  @NetworkTrafficDump
+  @Test(
+          dataProviderClass = FandomAdsDataProvider.class,
+          dataProvider = "vuapPage",
+          groups = {"AdsVuapFandomDesktop", "AdsVuapVideoPauseFandom"}
+  )
+  public void adsVuapVideoPausesFandom(String pageType, String pageName, String slotName) {
+    AdsFandomObject fandomPage = loadPage(pageName, pageType);
+    AutoplayVuap videoFanTakeover = prepareSlot(slotName, fandomPage);
+
+    networkTrafficInterceptor.startIntercepting();
+
+    videoFanTakeover.play();
+    videoFanTakeover.waitForFirstQuartile(networkTrafficInterceptor);
+    videoFanTakeover.togglePause();
+    double time = videoFanTakeover.getCurrentTime();
+
+    try {
+      TimeUnit.SECONDS.sleep(3);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)